src/components/hooks/use-signin-and-signup.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from 'react'
+import apiManager from 'common/websocket'
+import { isBrowser } from 'common/utility'
+
+export const useSigninAndSignup = (platform: string, token: string) => {
+    const [loading, setLoading] = useState(false)
+    const [has_account, setHasAccount] = useState(null)
+    const [authorized, setAuthorized] = useState(false)
+    const [account_error, setAccountError] = useState(null)
+    const [create_account_error, setCreateAccountError] = useState(null)
+    const [service_token, setServiceToken] = useState(null)
+
+    useEffect(() => {
+        if (token) {
+            setLoading(true)
+            apiManager
+                .augmentedSend('authorize', {
+                    authorize: token,
+                })
+                .then(() => {
+                    setAuthorized(true)
+                })
+                .catch((reason) => {
+                    setAccountError(reason.error.message)
+                    setLoading(false)
+                })
+        } else {
+            setAccountError('Token in URL is required as token1=')
+        }
+    }, [token])
+
+    useEffect(() => {
+        if (authorized) {
+            apiManager
+                .augmentedSend('trading_platform_accounts', {
+                    trading_platform_accounts: 1,
+                    platform,
+                })
+                .then((response) => {
+                    if (response.trading_platform_accounts.length > 0) {
+                        setHasAccount(true)
+                    } else {
+                        setHasAccount(false)
+                    }
+                })
+                .catch((reason) => {
+                    setAccountError(reason.error.message)
+                    setLoading(false)
+                })
+        }
+    }, [authorized, platform])
+
+    useEffect(() => {
+        if (has_account === false) {
```

Comment:
```suggestion
        if (!has_account) {
```

Reply:
It doesn't work that way because the default value is `null` and that condition should only work when we fetch data and the value is `false` instead of `null`.